tests/integration/test_examples.py

```diff
@@ -123,6 +123,55 @@ class ExampleIntegrationTestConfiguration(NamedTuple):
     ExampleIntegrationTestConfiguration(
         name="whylogs", validation_function=whylogs_example_validation
     ),
+    ExampleIntegrationTestConfiguration(
+        name="statistics",
+        validation_function=generate_basic_validation_function(
+            pipeline_name="boston_housing_pipeline",
+            step_count=3,
+        ),
+    ),
+    ExampleIntegrationTestConfiguration(
+        name="lineage",
+        validation_function=generate_basic_validation_function(
+            pipeline_name="boston_housing_pipeline",
+            step_count=4,
+        ),
+    ),
+    ExampleIntegrationTestConfiguration(
+        name="dag_visualizer",
+        validation_function=generate_basic_validation_function(
+            pipeline_name="boston_housing_pipeline",
+            step_count=3,
+        ),
+    ),
+    ExampleIntegrationTestConfiguration(
+        name="standard_interfaces",
+        validation_function=generate_basic_validation_function(
+            pipeline_name="TrainingPipeline",
+            step_count=6,
+        ),
+    ),
+    ExampleIntegrationTestConfiguration(
+        name="airflow_local",
+        validation_function=generate_basic_validation_function(
+            pipeline_name="mnist_pipeline",
+            step_count=4,
+        ),
+    ),
+    ExampleIntegrationTestConfiguration(
+        name="class_based_api",
```

Comment:
'm curious, how does the test framework know which file to run for this example ? There is no run.py file, but a bunch of chapter_*.py files.

Reply:
Ok, I think I got it, the function that selects which of the python files in an example is in zenml.cli.example:

```
    @property
    def executable_python_example(self) -> str:
        """Return the Python file for the example"""
        if self.has_single_python_file:
            return self.python_files_in_dir[0]
        elif self.run_dot_py_file:
            return self.run_dot_py_file
        elif self.has_any_python_file:
            logger.warning(
                "This example has multiple executable python files. "
                "The last one in alphanumerical order is taken."
            )
            return sorted(self.python_files_in_dir)[-1]
        else:
            raise RuntimeError(
                "No pipeline runner script found in example. "
                f"Files found: {self.python_files_in_dir}"
            )
```

In our case, the last one in alphanum order is executed.

Reply:
The logic for this is in `src/zenml/cli/example.py`. We first check whether there is just one python file inside the example folder. If so, that's the one we run. Then if there are multiple files but one of them is called `run.py`, then that's the one we run. And finally if there are multiple files (but none of them are called `run.py`, when we just pick the last one in alphanumerical order to run.